### PR TITLE
Add login persistence (Locally + Firebase)

### PR DESCRIPTION
**Pull request by lumapools**
The app will, from now on, remember which user is logged in (even if the user quits the app) and link the account with Firebase (add the public information of the user to the database) and the User Profile Fragment. The picture of the user is shown on the profile.
Left: Account page (login action + status), 
Right: User Profile (with profile picture)
Webp net-resizeimage (4) Webp net-resizeimage (5)

### CODE REVIEW

File: app/src/main/java/com/example/sharingang/ItemsListFragment.kt [hunk 1]
```diff
@@ -42,8 +43,20 @@ class ItemsListFragment : Fragment() {
...
+                getString(R.string.preference_user_info),
+                Context.MODE_PRIVATE
+            )
             val action =
+                if (sharedPreferences == null) {
                     ItemsListFragmentDirections.actionItemsListFragmentToUserProfileFragment("test")
+                } else {
+                    ItemsListFragmentDirections.actionItemsListFragmentToUserProfileFragment(
+                        sharedPreferences.getString(
+                            getString(R.string.account_firebase_uid), "test"
```
Comment (bafavre) [inline app/src/main/java/com/example/sharingang/ItemsListFragment.kt:56]: Should we keep "test" as a default string? I don't know what else to put but maybe test isn't the best.

File: app/src/main/java/com/example/sharingang/UserProfileFragment.kt [hunk 2]
```diff
@@ -33,10 +33,11 @@ class UserProfileFragment : Fragment() {
         viewModel.setUser(args.userId)
         binding.viewModel = viewModel
 
+        /*
         lifecycleScope.launch(Dispatchers.IO) {
             viewModel.createTestUser(args.userId)
         }
-
+        */
```
Comment (bafavre) [inline app/src/main/java/com/example/sharingang/UserProfileFragment.kt:40]: Comments should be removed

File: app/src/main/java/com/example/sharingang/UserProfileViewModel.kt [hunk 3]
```diff
@@ -27,7 +27,7 @@ class UserProfileViewModel @Inject constructor(
     fun setUser(userId: String) {
         _userId.postValue(userId)
     }
-
+    /*
     // Only for testing
     suspend fun createTestUser(userId: String) {
         if (userRepository.get(userId) == null) {
```
Comment (bafavre) [inline app/src/main/java/com/example/sharingang/UserProfileViewModel.kt:33]: Maybe remove the comments

**Review by bafavre — CHANGES_REQUESTED** [review]
Good job, only a few remarks :)
Also maybe consider using DataStore instead of SharedPreferences : https://developer.android.com/topic/libraries/architecture/datastore

File: app/src/main/java/com/example/sharingang/AccountFragment.kt [hunk 4]
```diff
@@ -97,4 +145,78 @@ class AccountFragment : Fragment() {
...
+                    lifecycleScope.launch(Dispatchers.IO) {
+                        userRepository.add(
+                            User(
+                                id = user?.uid,
+                                name = user?.displayName!!,
+                                profilePicture = user.photoUrl!!.toString()
+                            )
+                        )
+                    }
+                    editor.putString(getString(R.string.account_firebase_uid), user?.uid)
```
Comment (betrisey) [inline app/src/main/java/com/example/sharingang/AccountFragment.kt:185]:
```suggestion
                if (task.isSuccessful && auth.currentUser != null) {
                    val user = auth.currentUser!!
                    lifecycleScope.launch(Dispatchers.IO) {
                        userRepository.add(
                            User(
                                id = user.uid,
                                name = user.displayName!!,
                                profilePicture = user.photoUrl?.toString()
                            )
                        )
                    }
                    editor.putString(getString(R.string.account_firebase_uid), user.uid)
```
I think it's better to check if the user is not null before. The code becomes a bit simpler inside and we avoid adding null user to the database.

File: app/src/main/java/com/example/sharingang/ItemsListFragment.kt (hunk 1 above)
Comment (betrisey) [inline app/src/main/java/com/example/sharingang/ItemsListFragment.kt:56]: If the user is not logged in, I think we shouldn't go to the user profile at all, maybe display a message with a Toast or Snackbar.
Also for the tests, I'll try to see if we can use dependency injection to get a fake user ID for the tests.

**Review by betrisey — CHANGES_REQUESTED** [review]

**Review by betrisey — APPROVED** [review]

**Review by bafavre — APPROVED** [review]

**Pull request merged**